plugin/src/test/java/org/jenkinsci/plugins/workflow/cps/CpsScriptTest.java

```diff
@@ -92,4 +92,124 @@ public void evaluateShallSandbox() throws Exception {
         r.assertLogContains("Scripts not permitted to use method groovy.lang.Script run java.io.File java.lang.String[]", b);
     }
 
+    @Test public void methodTooLargeExceptionFabricated() throws Exception {
+        // Fabricate a MethodTooLargeException which "normally" happens when evaluated
+        // groovy script becomes a Java class too large for Java to handle internally.
+        // In Jenkins practice this can happen not only due to large singular pipelines
+        // (one big nudge to offload code into shared libraries), but was also seen due
+        // to heavy nesting of exception handling and other loops (simple refactoring
+        // can help).
+        WorkflowJob p = r.createProject(WorkflowJob.class);
+        // sandbox == false to allow creation of the exception here:
+        p.setDefinition(new CpsFlowDefinition(
+                "import groovyjarjarasm.asm.MethodTooLargeException;\n\n" +
+                        "throw new MethodTooLargeException('className', 'methodName', 'methodDescriptor', 65535);"
+                , false));
+        WorkflowRun b = r.buildAndAssertStatus(Result.FAILURE, p);
+        r.assertLogContains("groovyjarjarasm.asm.MethodTooLargeException: Method too large: className.methodName methodDescriptor", b);
+        r.assertLogContains("at WorkflowScript.run(WorkflowScript:3)", b);
+        r.assertLogContains("at ___cps.transform___(Native Method)", b);
+    }
+
+    @Test public void methodTooLargeExceptionRealistic() throws Exception {
+        // See comments above. Here we try to really induce a "method too large"
+        // condition by abusing the nesting of exception-handling, too many stages
+        // or methods, and whatever else we can throw at it.
+        WorkflowJob p = r.createProject(WorkflowJob.class);
+        StringBuffer sbMethods = new StringBuffer();
+        StringBuffer sbStages = new StringBuffer();
+
+        // Limits to the "max":
+        // * java.lang.StackOverflowError varies per JDK platform
+        //   (CI on Linux was unhappy with 255, on Windows with 1023)
```

Comment:
You _can_ set a per-thread stack size, but unfortunately the unit of measurement is unspecified and presumably specific to JVM implementation details, so this is probably not that helpful.

Reply:
(Why would you would be getting a JVM-level SOE, anyway? This is all dealing with the CPS VM that does not use the native stack.)

Reply:
Saw some code in this plugin throwing it (for "excessively nested" cases at least), so thought it is worth keeping in mind.